lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -1180,6 +1180,31 @@ LogicalResult OrOp::canonicalize(OrOp op, PatternRewriter &rewriter) {
     return success();
   }
 
+  // or(mux(c_1, a, 0), mux(c_2, a, 0), ..., mux(c_n, a, 0)) -> mux(or(c_1, c_2,
+  // .., c_n), a, 0)
+  if (auto firstMux = op.getOperand(0).getDefiningOp<comb::MuxOp>()) {
+    APInt value;
+    SmallVector<Value> conditions;
+    auto check = [&](Value v) {
+      auto mux = v.getDefiningOp<comb::MuxOp>();
+      if (!mux)
+        return false;
+      conditions.push_back(mux.getCond());
+      return firstMux.getTwoState() &&
+             firstMux.getTrueValue() == mux.getTrueValue() &&
+             matchPattern(mux.getFalseValue(), m_ConstantInt(&value)) &&
+             value.isZero();
```

Comment:
zero can be checked against firstMux outside the lambda/loop.  then just check firstMux.getFalseValue() == mux.getFalseValue().